src/item.cpp

```diff
@@ -8029,7 +8029,8 @@ ret_val<bool> item::is_gunmod_compatible( const item &mod ) const
         return ret_val<bool>::make_failure( _( "cannot have a brass catcher" ) );
 
     } else if( mod.type->gunmod->location.name() == "magazine" ||
-               mod.type->gunmod->location.name() == "mechanism" ) {
+               mod.type->gunmod->location.name() == "mechanism" ) &&
+               ( ammo_remaining() > 0 || magazine_current() ) ){
```

Comment:
```suggestion
               mod.type->gunmod->location.name() == "mechanism" &&
               ( ammo_remaining() > 0 || magazine_current() ) ) {
```
I'm using this as an excuse to test suggestions since I've never used this feature ;)

Reply:
> I'm using this as an excuse to test suggestions since I've never used this feature ;)

:)